examples/offline_inference_baichuan.py

```diff
@@ -0,0 +1,145 @@
+            if message["role"] == "user":
+                round_tokens.append(user_token_id)
+            else:
+                round_tokens.append(assistant_token_id)
+            round_tokens.extend(tokenizer.encode(message["content"]))
+        if len(history_tokens) == 0 or len(history_tokens) + len(
+                round_tokens) <= max_history_tokens:
+            history_tokens = round_tokens + history_tokens  # concat left
+            if len(history_tokens) < max_history_tokens:
+                continue
+        break
+
+    input_tokens = system_tokens + history_tokens
+    if messages[-1]["role"] != "assistant":
+        input_tokens.append(assistant_token_id)
+    input_tokens = input_tokens[-max_input_tokens:]  # truncate left
+    return input_tokens
+
+
+# Sample prompts.
+prompts = [
+    [{
+        "role": "user",
+        "content": "Hello, my name is"
+    }],
+    [{
+        "role": "user",
+        "content": "The president of the United States is"
+    }],
+    [{
+        "role": "user",
+        "content": "The capital of France is"
+    }],
+    [{
+        "role": "user",
+        "content": "The future of AI is"
+    }],
+]
+
+# Taken from generation_config.
+sampling_params = SamplingParams(temperature=0.3,
+                                 top_k=5,
+                                 top_p=0.85,
+                                 max_tokens=2048,
+                                 stop=[".</s>"]
+                                 # repetition_penalty=1.05
+                                 )
+
+# Create an LLM.
+llm = LLM(model="baichuan-inc/Baichuan2-13B-Chat",
+          trust_remote_code=True,
+          tensor_parallel_size=2,
+          tokenizer_mode='slow')
+tokenizer = AutoTokenizer.from_pretrained("baichuan-inc/Baichuan2-13B-Chat",
+                                          use_fast=False,
+                                          trust_remote_code=True)
+# bypass chat format issue with explicit tokenization
+prompt_token_ids = [
+    build_chat_input(tokenizer, prompt, max_new_tokens=2048)
+    for prompt in prompts
+]
+# Generate texts from the prompts. The output is a list of RequestOutput objects
+# that contain the prompt, generated text, and other information.
+outputs = llm.generate(prompt_token_ids=prompt_token_ids,
+                       sampling_params=sampling_params)
+# Print the outputs.
+for output in outputs:
+    prompt = output.prompt
+    generated_text = output.outputs[0].text
+    print(f"Prompt: {prompt!r},\n Generated text: {generated_text!r}")
+"""
+Sample output
+Processed prompts: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:13<00:00,  1.93s/it]
+Prompt: None,
+ Generated text: 'Hello, my name is [your name]. Nice to meet you!'
+Prompt: None,
+ Generated text: 'The current president of the United States is Joe Biden, who was sworn into office on January 20, 2021.'
+>>>>>> Baichuan2-13B-Chat 8Bit Demo: 
```

Comment:
What is the meaning of 8Bit here? I see that the model you loaded above is `Baichuan2-13B-Chat`.

Reply:
They're just for comparing vllm's output with HF model's output. A consistency check.

Reply:
okay